fe/fe-core/src/main/java/org/apache/doris/nereids/rules/exploration/mv/InitMaterializationContextHook.java

```diff
@@ -129,15 +131,24 @@ private List<MaterializationContext> createAsyncMaterializationContext(CascadesC
             Set<TableIf> usedTables) {
         Set<MTMV> availableMTMVs = getAvailableMTMVs(usedTables, cascadesContext);
         if (availableMTMVs.isEmpty()) {
-            LOG.debug(String.format("Enable materialized view rewrite but availableMTMVs is empty, current queryId "
-                    + "is %s", cascadesContext.getConnectContext().getQueryIdentifier()));
+            LOG.debug("Enable materialized view rewrite but availableMTMVs is empty, current queryId "
+                    + "is {}", cascadesContext.getConnectContext().getQueryIdentifier());
             return ImmutableList.of();
         }
         List<MaterializationContext> asyncMaterializationContext = new ArrayList<>();
         for (MTMV materializedView : availableMTMVs) {
             MTMVCache mtmvCache = null;
             try {
                 mtmvCache = materializedView.getOrGenerateCache(cascadesContext.getConnectContext());
+                // If mv property use_for_rewrite is set false, should not partition in
+                // query rewrite by materialized view
+                String usedForRewrite = materializedView.getMvProperties().get(
+                        PropertyAnalyzer.PROPERTIES_USE_FOR_REWRITE);
+                if (!StringUtils.isEmpty(usedForRewrite) && !Boolean.parseBoolean(usedForRewrite)) {
```

Comment:
Suggest extracting a method from the logic for obtaining usedForRewrite and placing it in MTMV, similar to getWorkload Group,

Reply:
have fixed